### PR TITLE
Fix sorting order bug.

### PR DESCRIPTION
**Pull request by stefanondisponibile**
Fixes sorting order bug causing documents order being changed by refined filter (`_merge_id_filter`).
This was confirmed by running the following two queries:

**query_1**
```
{
  "some_id": ObjectId("5630be57e0e2ee4069000018")
}
```

**query_2**
```
{
    "$and": [
      {"some_id": ObjectId("5630be57e0e2ee4069000018")}, 
      {'_id': {'$gte': ObjectId('4f58d81ef16cd2000100009d'), '$lt': ObjectId('5da88d07f93d78156e0e0689')}}
    ]
}
```

**The two queries led to different ordering on the same subset of data.**

------------------------

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by aaltay** [comment]
R: @y1chi

**Comment by y1chi** [comment]
Do these two query return different sets of data or they just mutate the documents ordering in same set?

As beam is inherently for distributed processing I think the ordering is generally not guaranteed.

**Comment by stefanondisponibile** [comment]
@y1chi the 2 queries return a different ordering of the same set of data. Preserving order is not a concern (for me) in this case, but it seems it is for Beam, as an Error is raised on the ‘try_claim‘.

**Comment by y1chi** [comment]
I see, try_claim does seem to assume in range order processing, thanks for the catch.

Could you provide a more meaningful commit message to mention python sdk mongodbio?

File: sdks/python/apache_beam/io/mongodbio.py [hunk 1]
```diff
@@ -192,7 +192,7 @@ def get_range_tracker(self, start_position, stop_position):
   def read(self, range_tracker):
     with MongoClient(self.uri, **self.spec) as client:
       all_filters = self._merge_id_filter(range_tracker)
-      docs_cursor = client[self.db][self.coll].find(filter=all_filters)
+      docs_cursor = client[self.db][self.coll].find(filter=all_filters).sort("_id")
```
Comment (y1chi) [inline sdks/python/apache_beam/io/mongodbio.py]: if you don't want to fix the unit tests I think you can use .sort([('_id', ASCENDING)]) so it is more compliant with the fake cursor in unit test.

File: sdks/python/apache_beam/io/mongodbio.py [hunk 2]
```diff
@@ -192,7 +192,7 @@ def get_range_tracker(self, start_position, stop_position):
   def read(self, range_tracker):
     with MongoClient(self.uri, **self.spec) as client:
       all_filters = self._merge_id_filter(range_tracker)
-      docs_cursor = client[self.db][self.coll].find(filter=all_filters).sort("_id")
+      docs_cursor = client[self.db][self.coll].find(filter=all_filters).sort([('_id', ASCENDING)])
```
Comment (y1chi) [inline sdks/python/apache_beam/io/mongodbio.py]: Probably need to break into two lines since it is violating line length lint rule.

**Comment by y1chi** [comment]
Run Python MongoDBIO_IT

**Comment by y1chi** [comment]
@stefanondisponibile fyi the lint is still failing for wrong indentation.

**Comment by y1chi** [comment]
I think there is a redundant square bracket.

**Comment by y1chi** [comment]
Run Python PreCommit

**Comment by y1chi** [comment]
@chamikaramj Cham could you help to merge.

**Comment by y1chi** [comment]
@aaltay Ahmet could you help to merge

**Pull request merged**